doc/docs/Python_User_Interface.md

```diff
@@ -87,40 +87,40 @@ control various parameters of the Meep computation.
 
 ```python
 def __init__(self,
-             cell_size,
-             resolution,
-             geometry=None,
-             sources=None,
-             eps_averaging=True,
-             dimensions=3,
-             boundary_layers=None,
-             symmetries=None,
-             force_complex_fields=False,
-             default_material=Medium(),
-             m=0,
-             k_point=False,
+             cell_size: meep.geom.Vector3 = None,
```

Comment:
Remember that these arguments can be tuples instead of `Vector3`, and they get converted to `Vector3` internally

Reply:
(Note that unions of types can be expressed as `Union[X, Y]` if you do `from typing import Union`).

Reply:
Note also that you presumably need to allow `NoneType` in order to accept a `None` argument.

Reply:
You can wrap the type with `Optional` to allow for `None`, e.g. `Optional[Vector3] = None`

Reply:
I have wrapped the type of several of the constructor arguments which have `None` as the default with `Optional`. Also, I introduced a `Vector3Type` alias as `Union[Vector3, Tuple[float, ...], Tuple[int, ...]` which includes tuples.